lib/configuration/client_configuration.js

```diff
@@ -110,38 +115,56 @@ function ConfigurationUtil(fsPromisesModule, processModule) {
    * @return {Promise<ClientConfig>} Client configuration.
    */
   this.getClientConfig = async function (configFilePath) {
+    Logger.getInstance().debug('Retrieving client config');
+
     const path = await findConfig(configFilePath);
     if (!exists(path) || path === '') {
+      Logger.getInstance().info('No config file path found. Client config will not be used.');
       return null;
     }
 
     const isFileOk = await isFileNotWritableByGroupOrOthers(path, fsPromises).catch(err => {
+      Logger.getInstance().warn('Failed to inspect config file path permissions. Client config will not be used.');
       throw new ConfigurationError('Finding client configuration failed', err);
     });
 
     if (!isFileOk) {
+      Logger.getInstance().warn(`Config file path permissions are invalid. File: ${path} can be modified by group or others. Client config will not be used.`);
       throw new ConfigurationError(`Configuration file: ${path} can be modified by group or others`, 'IncorrectPerms');
     }
+    Logger.getInstance().debug(`Config file path permissions are valid. Path: ${path}.`);
 
     const configFileContents = await readFileConfig(path);
+    Logger.getInstance().info('Using client configuration from path: %s.', path);
+
     return configFileContents == null ? null : parseConfigFile(path, configFileContents);
   };
 
   function readFileConfig(filePath) {
+    Logger.getInstance().debug(`Reading config file. Path: ${filePath}`);
+
     if (!filePath) {
+      Logger.getInstance().trace(`Path of config file is not specified. Nothing to read. Path: ${filePath}`);
       return Promise.resolve(null);
     }
     return fsPromises.readFile(filePath, { encoding: 'utf8' })
       .catch(err => {
+        Logger.getInstance().debug(`Reading configuration from the file failed. Path: ${filePath}`);
         throw new ConfigurationError('Finding client configuration failed', err);
       });
   }
 
   function parseConfigFile(path, configurationJson) {
+    Logger.getInstance().debug('Parsing config file: %s.', path);
     try {
       const parsedConfiguration = JSON.parse(configurationJson);
+      Logger.getInstance().trace('Config file contains correct JSON structure. Validating the input.');
+
       checkUnknownEntries(parsedConfiguration);
       validate(parsedConfiguration);
+
+      Logger.getInstance().debug('Config file contains valid configuration input.');
+
       return new ClientConfig(
```

Comment:
We have to log the level and the path on the info level that is used to configure the logger.

Reply:
Nice catch! Added 👍